### PR TITLE
Hao/column drag drop modfication

### PR DESCRIPTION
**Pull request by haophn01**
About Column Drag and Drop modification:
- Job card that can move up and down with transition is ease for the user experience, when each job card is move and down within the column, on the index will be update, the content of job card wont be change (id, status, etc). 
- If a job card is move to other stages, only the index and job stage status will be update.
- There was an additional request from you that you want the border color change between "from" and "to" column when we perform drag and drop. 
- Also I fixed the page view which remove the extra blank space at bottom. 
- Each column can hold up to 8 cards in the bucket, if it is over, user will have to perform scrolling if they want to add more. Don't worry the column itself does not expand as the card is added more. Think it as a virtual list.

### CODE REVIEW

File: client/src/lib/store/track.ts [hunk 1]
```diff
@@ -29,36 +29,62 @@ interface TrackState {
...
-      // removes application
       state[from] = state[from].filter(
         (application) => application.id !== applicationId
       );
+      return { ...state };
+    }),
 
-      // moves application to new stage
-      state[to] = [movedApplication!, ...state[to]];
+  moveApplication: (from, to, applicationId, sourceIndex, destinationIndex) =>
```
Comment (MekhiHart) [inline client/src/lib/store/track.ts:67]: why did the logic for the moveApplication have to be changed including the parameters?

**Review by MekhiHart — CHANGES_REQUESTED** [review]
When the card is being dragged, let's only highlight the column blue when it's in a new column. 
Screenshot 2024-10-27 at 8 35 28 AM

File: client/src/lib/store/track.ts (hunk 1 above)
Comment (haophn01) [reply client/src/lib/store/track.ts:67]: This allow the user to reorder the job card within the column base on the job they want to prioritize monitor.

Comment (MekhiHart) [reply client/src/lib/store/track.ts:67]: Got it

**Review by MekhiHart — APPROVED** [review]

**Review by MekhiHart — APPROVED** [review]

**Pull request merged**